### PR TITLE
Add deployment directory with example configuration for testing

### PR DESCRIPTION
**Pull request by acmiyaguchi**
This adds configuration for running integration tests. It's a bit difficult to reference the files that have been deployed by data-sre, so this generates the configuration directly via terraform. I hope to use this structure for testing changes in major revisions of the container.

### CODE REVIEW

File: deployment/testing-v3/compose/admin/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+version: "3.4"
+
+services:
+  app:
+    image: mozilla/prio-processor:v3.1.1
```
Comment (acmiyaguchi) [inline deployment/testing-v3/compose/admin/docker-compose.yml:5]: I've pinned the containers here for the sake of reproducibility. I'm going to break a lot of the current conventions in order to support #112

**Comment by acmiyaguchi** [comment]
Just a note that terraform was a joy to use in this context because of how the servers need to be configured. Each server gets a bucket (private and shared), and has to authorize three service accounts (the admin, itself, and the partner). Being able to run `terraform apply` to apply changes to the infrastructure is nice. The service accounts are scoped to only the resources that they need, which makes me comfortable having the private keys on my local disk since there is very little that can be abused with the buckets and how they are managed.

I'm going to be using this configuration as a template for future changes which will involve provisioning HMAC keys instead of service account keys (which means I'll have to figure an alternative way to add the environment variable for the keys because they're actually sensitive). I also want to clean up the permissions a bit and create a third bucket for each server for ingestion. Having the terraform config around makes it easy to play with this.

I also had to make a custom content-test.json for the data configuration because iterating through each one is really slow. I think I may have to optimize the spark job to process everything in one go, otherwise this is going to be untenable for longer running jobs.

File: deployment/testing-v3/compose/admin/.env.template [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+APP_NAME=test-app
+DATA_CONFIG=/app/config/content.json
+ORIGIN_CONFIG=/app/config/telemetry_origin_data_inc.json
+BUCKET_PREFIX=test-app/v1
+
+# relative to the docker-compose file
+GOOGLE_APPLICATION_CREDENTIALS="../../.service-account-keys/service-account-admin-private-key.json"
+
+PUBLIC_KEY_HEX_INTERNAL=1723D28D922FD045A3EBECE5FA9BBD67DF28B60B5666203DA06E0CE296D7DF11
+BUCKET_INTERNAL_PRIVATE=gs://a-private-a82843a795cf9ef5
```
Comment (wlach) [inline deployment/testing-v3/compose/admin/.env.template:13]: Just confirming that we're ok with checking these into source control?

File: deployment/testing-v3/scripts/list-bucket [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+for bucket in $(echo "$buckets" | grep b-); do
+cat << EOF
+### \`$bucket\`
+
+\`\`\`
+$(sort_recursive_listing "$bucket")
+\`\`\`
+
+EOF
+done
```
Comment (wlach) [inline deployment/testing-v3/scripts/list-bucket:57]:
```suggestion
done

```

**Review by wlach — APPROVED** [review]
This looks pretty reasonable to me. Is the plan to run these integration tests via ci?

File: deployment/testing-v3/compose/admin/.env.template (hunk 2 above)
Comment (acmiyaguchi) [reply deployment/testing-v3/compose/admin/.env.template:13]: Yep, I wanted to have the .env.template have realistic values to be able to reference. There's nothing sensitive in these configuration files.

**Comment by acmiyaguchi** [comment]
CI is definitely a possibility, but probably not at this moment. I am planning on adding a minio example that won't require GCP access in the near future, so that's probably a good candidate for CI.

**Pull request merged**